data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -282,37 +294,75 @@ public void doOutput(final Collection<Record<Event>> records) {
 
       BulkOperation bulkOperation;
 
-      if (StringUtils.equalsIgnoreCase(action, BulkAction.CREATE.toString())) {
-
-        final CreateOperation.Builder<Object> createOperationBuilder = new CreateOperation.Builder<>()
-                .index(indexName)
-                .document(document);
-
-        docId.ifPresent(createOperationBuilder::id);
-        routing.ifPresent(createOperationBuilder::routing);
-
-        bulkOperation = new BulkOperation.Builder()
-                .create(createOperationBuilder.build())
-                .build();
+      String eventAction = action;
```

Comment:
This is significant logic. Can we pull it out into its own class? It will be easier to test this way and make this method more manageable.

Reply:
OK, will make it a different method.